qiskit_ibm_provider/ibm_backend_service.py

```diff
@@ -172,18 +172,37 @@ def backends(
                     )
                 backends.append(self._backends[backend_name])
         elif name:
+            default_instance = None
+            if not instance:
```

Comment:
I don't think this check is necessary since the outer `if` on line 162 would have been true if `instance` is not `None`, i.e. this `elif` only happens if `instance` is None.